src/core/lib/attribute-manager.js

```diff
@@ -310,13 +320,17 @@ export default class AttributeManager {
    * @return {Object} attributes - descriptors
    */
   getChangedAttributes({clearChangedFlags = false}) {
-    const {attributes} = this;
+    const {attributes, attributeTransitionManger} = this;
     const changedAttributes = {};
     for (const attributeName in attributes) {
       const attribute = attributes[attributeName];
       if (attribute.changed) {
         attribute.changed = attribute.changed && !clearChangedFlags;
-        changedAttributes[attributeName] = attribute;
+
+        // If there is transition, let the transition manager handle the update
```

Comment:
Seems a bit subtle to hide this logic in here. Maybe refactor this (split getChangedAttributes?) to make this more explicit.